### DISCUSSION TITLE
Accessing other fields from custom fieldtype index

### DISCUSSION THREAD

**Discussion by david-windsock** [statamic/cms#10019] [Q&A]
As described [here](https://statamic.dev/extending/fieldtypes#accessing-other-fields), accessing other fields to get its values from a form fieldtype is possible using this on the Vue component:

```
inject: ['storeName'],
 
computed: {
    formValues() {
        return this.$store.state.publish[this.storeName].values;
    },
},
```

But this doesn't work from index fieldtypes, throwing the following error:

```
TypeError: Cannot read properties of undefined (reading 'values')
```

Is there a way to do this? Thanks!

**Comment by david-windsock** [comment]
I found that `this.values` is defined but only shown values are available. Is there any way to get values from hidden fields on the index?.

**Comment by jasonvarga** [comment]
>  Is there any way to get values from hidden fields on the index?

No. For performance reasons, we only get the values from the server that will be used in the listing.

You can pass down what you need from PHP using the fieldtype's preProcessIndex method.

**Comment by david-windsock** [comment]
Thanks @jasonvarga, it worked for me. I can get entry values using `$this->field->parent()` on `Fieldtype::preProcessIndex()` method.
